tests/binaries/heap_find_fake_fast.c

```diff
@@ -10,17 +10,79 @@
  */
 
 #include <stdlib.h>
+#include <assert.h>
 
 void break_here(void) {}
 
 // Fake chunk size field with a set NON_MAIN_ARENA flag.
 // Enough space afterwards to ensure only this fake size field is a candidate.
 char fake_chunk[0x80] __attribute__((aligned(0x10))) = "XXXXXXXX\x7f";
 
-int main(void)
-{
+// This buffer will contain the fake chunk sizes
+unsigned long buf[64] __attribute__((aligned(0x10)));
+
+// This is the address we want the fake chunks to overlap with
+unsigned long target_address;
+
+void setup_mem(unsigned long size, unsigned distance) {
+    memset(buf, 0, sizeof(buf));
+    target_address = 0;
+
+    char *chunk_size_addr = (char*)&target_address - distance + __SIZEOF_POINTER__;
+    *(unsigned long*)chunk_size_addr = size;
+}
+
+int main(void) {
+    assert((unsigned long)&target_address - (unsigned long)buf == sizeof(buf));
     // Initialize malloc so heap commands can run.
     void* m = malloc(0x18);
 
+    // A valid aligned fastbin chunk with no flags set
+    setup_mem(0x20, 0x10);
+    break_here();
+
+    // A valid aligned fastbin chunk with all flags set
+    setup_mem(0x2F, 0x10);
+    break_here();
+
+    // A valid unaligned fastbin chunk
+    setup_mem(0x20, 0x11);
+    break_here();
+
+    // A valid fastbin chunk that's too close to the target address
+    setup_mem(0x20, 0x8);
```

Comment:
Yea, this is testing that if the `size` field is actually on the target address, we don't return it. Just in case there's an off-by-one bug somewhere.